keras/callbacks.py

```diff
@@ -1003,9 +1003,13 @@ class LambdaCallback(Callback):
         # Plot the loss after every epoch.
         import numpy as np
         import matplotlib.pyplot as plt
-        plot_loss_callback = LambdaCallback(
-            on_epoch_end=lambda epoch, logs: plt.plot(np.arange(epoch),
-                                                      logs['loss']))
+
+        def plot_loss(epoch, logs, losses=[]):
+            losses.append(logs['loss'])
+            plt.plot(np.arange(epoch + 1), losses)
+            plt.show()  # Use the best method to update the plot for your backend
```

Comment:
@fchollet the proposed fix is actually creating a plot per epoch which is somewhat suboptimal but better than what we had before 

the shortest code snippet I could come up with that uses an interactive matplotlib backend and can be updated with new data looks like this:

```Python
import numpy as np
import matplotlib as mpl

# Use an interactive backend to get continous updates.
matplotlib.use('TkAgg') 
import matplotlib.pyplot as plt

plt.ion()
fig, ax = plt.subplots()
ax.set_xlim(0, num_epochs)
ax.set_ylim(0, 10)
loss_line, = ax.plot([], [])

def plot_loss(epoch, logs):
    loss_line.set_xdata(np.append(loss_line.get_xdata(), epoch))
    loss_line.set_ydata(np.append(loss_line.get_ydata(), logs['loss']))
    fig.canvas.draw_idle()
    fig.canvas.flush_events()

plot_loss_callback = LambdaCallback(on_epoch_end=plot_loss)

...
```

IMO it's a little too much code for a lambda but what do you think?